crates/oxc_linter/src/rules/promise/valid_params.rs

```diff
@@ -0,0 +1,190 @@
+use oxc_ast::{
+    ast::{CallExpression, Expression},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn valid_params_diagnostic(span0: Span, x0: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("eslint-plugin-promise(valid-params): {x0}")).with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct ValidParams;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforces the proper number of arguments are passed to Promise functions.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Calling a Promise function with the incorrect number of arguments can lead to unexpected
+    /// behavior or hard to spot bugs.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// Promise.resolve(1, 2)
+    /// ```
+    ValidParams,
+    correctness,
+);
+
+fn is_promise(call_expr: &CallExpression) -> bool {
+    let Some(member_expr) = call_expr.callee.get_member_expr() else {
+        return false;
+    };
+
+    let Some(prop_name) = member_expr.static_property_name() else {
+        return false;
+    };
+
+    // hello.then(), hello.catch(), hello.finally()
+    if matches!(prop_name, "then" | "catch" | "finally") {
+        return true;
+    }
+
+    // foo().then(foo => {})
+    if let Expression::CallExpression(inner_call_expr) = member_expr.object() {
+        return is_promise(inner_call_expr);
+    }
```

Comment:
I believe this is nonsense, dropping it still makes the test pass. Also this check is recursive so likely not great :)

It's adapted from, which honestly I don't fully understand :smile: 
https://github.com/eslint-community/eslint-plugin-promise/blob/main/rules/lib/is-promise.js#L26

Reply:
As we check all call expression in our rule, we will check twice for `doThing.then().anything(...)`. One for `doThing.then().anything(...)`, it is early bailout, another is `doThing.then` which report the error.

I also think it's safe to remove the recursive callback, since we can always visit the inner promise function call. =.=